x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+
+  await poll(async () => {
+    return axios
+      .post(
+        `${baseUrl}/api/v1/serverless/projects/security/${environmentId}/_reset-credentials`,
+        {},
+        {
+          headers: {
+            Authorization: `ApiKey ${apiKey}`,
+          },
+        }
+      )
+      .then((response) => {
+        credentials.password = response.data.password;
+        credentials.username = response.data.username;
+      })
+      .catch((error) => {
+        throw new Error(`${error.code}:${error.data}`);
+      });
+  });
+  return credentials;
+}
+
+// Wait until elasticsearch status goes green
+async function waitForEsStatusGreen(esUrl: string, auth: string, runnerId: string) {
+  await poll(async () => {
+    await axios
+      .get(`${esUrl}/_cluster/health?wait_for_status=green&timeout=50s`, {
+        headers: {
+          Authorization: `Basic ${auth}`,
+        },
+      })
+      .then((response) => {
+        log.info(`${runnerId}: Elasticsearch is ready with status ${response.data.status}.`);
+      })
+      .catch((error) => {
+        if (error.code === 'ENOTFOUND') {
+          log.info(`${runnerId}: The elasticsearch url is not yet reachable. Retrying in 20s...`);
+        }
+        throw new Error(`${runnerId} - ${error.code}:${error.data}`);
+      });
+  });
+}
+
+// Wait until Kibana is available
+async function waitForKibanaAvailable(kbUrl: string, auth: string, runnerId: string) {
+  await poll(async () => {
+    await axios
+      .get(`${kbUrl}/api/status`, {
+        headers: {
+          Authorization: `Basic ${auth}`,
+        },
+      })
+      .then((response) => {
+        if (response.data.status.overall.level !== 'available') {
+          log.info(`${runnerId}: Kibana is not available. Retrying in 20s...`);
+          throw new Error(`${runnerId}: Kibana is not available. Retrying in 20s...`);
+        }
+      })
+      .catch((error) => {
+        if (error.code === 'ENOTFOUND') {
+          log.info(`${runnerId}: The kibana url is not yet reachable. Retrying in 20s...`);
+        }
+        throw new Error(`${runnerId} - ${error.code}:${error.data}`);
+      });
+  });
+}
+
+export const cli = () => {
+  run(
+    async () => {
+      log = new ToolingLog({
+        level: 'info',
+        writeTo: process.stdout,
+      });
+
+      const PROJECT_NAME_PREFIX = 'kibana-cypress-security-solution-ephemeral';
+      const QA_BASE_ENV_URL = 'https://global.qa.cld.elstc.co';
+
+      // Checking if API key is either provided via env variable or in ~/.elastic.cloud.json
+      if (!process.env.CLOUD_QA_API_KEY && !getApiKeyFromElasticCloudJsonFile()) {
+        log.error('The api key for the environment needs to be provided with the env var API_KEY.');
+        log.error(
+          'If running locally, ~/.elastic/cloud.json is attempted to be read which contains the api key.'
+        );
+        // eslint-disable-next-line no-process-exit
+        return process.exit(0);
```

Comment:
I'm curious if just return a number. Will it work?

Reply:
Still normally exits the runtime. You found a really really good point. At the moment process.exit(0) gives us correct execution...........

We need to fix it with process.exit(1) in order to get a failure in buildkite. Very **VERY GOOD** prompt!

Reply:
https://github.com/elastic/kibana/pull/169311/commits/3006b32f33a4dfdab732e9a34e60054ac69b4bef